sample/rules.go

```diff
@@ -159,15 +162,21 @@ func ruleMatchesTrace(t *types.Trace, rule *config.RulesBasedSamplerRule, checkN
 				matched++
 				continue
 			} else {
-				// if HasRootSpan is one of the conditions and it didn't match,
```

Comment:
why remove this comment? it explains the return false.

Reply:
Yeah I think I overcorrected here. I'll add it back.